app/helpers/application_helper.rb

```diff
@@ -512,6 +512,14 @@ def show_image_name(image_field_name)
     return ""
   end
 
+  def number_of_notes(student)
+    if student.notes.count > 0
+      student.notes.count
+    else
+      ""
```

Comment:
This function returns count ( Integer). You can return 0 in the else condition.

Reply:
We don't need to return 0, because we display this return on the students index page in the Notes column. We show the number of notes or nothing. I don't think we should show the number zero if there are no notes.

Reply:
In that case, we are still building a link when else condition is true and value is returned as blank. We can refactor the code by making the link code conditional in **app/views/programs/students/_student_list.html.erb**. This way the link is built and displayed only when count is greater than 0.

Reply:
I agree with that, I'll fix the logic.

Reply:
I created links to students' pages only if students have notes.